Add tests for labeling Backups

ttpcodes opened the pull request:
This PR adds tests for the ability to add `Labels` to `Backups`, a feature that was requested in longhorn/longhorn#563 and implemented in longhorn/longhorn-manager#335. There are a total of six tests implemented by this PR. These are the base four:
- `test_backup_labels`: Tests that creating a `Backup` applies the `Labels` specified by the user.
- `test_backup_kubernetes_status`: Tests that the `KubernetesStatus` is set and restored properly when a user creates a `Backup`.
- `test_recurring_job_labels`: Tests that a `RecurringJob` applies user specified `Labels` and the appropriate `Label: RecurringJob`.
- `test_recurring_job_kubernetes_status`: Tests that a `RecurringJob` properly stores the latest `KubernetesStatus`.

Additionally, two tests have `BaseImage` variants to ensure that the appropriate `Labels` related to `BaseImages` are applied.
- `test_backup_labels_with_base_image`: Runs `test_backup_labels`, but on a `Volume` with an assigned `BaseImage`.
- `test_recurring_job_labels_with_base_image`: Runs `test_recurring_job_labels`, but on a `Volume` with an assigned `BaseImage`.

As an aside: `test_recurring_job_labels_with_base_image` currently fails as a result of longhorn/longhorn#630.

yasker commented on manager/integration/tests/test_kubernetes.py:
```diff
@@ -360,3 +364,73 @@ def test_pvc_creation_with_default_sc_set(
+        'namespace': '',
+        'pvcName': '',
+        'lastPVCRefAt': b["snapshotCreated"],
+        'lastPodRefAt': b["snapshotCreated"],
```
The pod wasn't set, so `lastPodRefAt` shouldn't be set in this case.

ttpcodes commented:
- Modified tests to properly handle the cases when `LastPodRefAt` and `LastPVCRefAt` should and shouldn't be set.
- Added a test case for ensuring the `Pod`-related fields of `KubernetesStatus` are stored properly.

ttpcodes commented:
- Modified tests to handle the cases of `LastPodRefAt` and `LastPVCRefAt` being set and unset.
